pkg/utils/configMapUtils_test.go

```diff
@@ -0,0 +1,148 @@
+	experiment.JobName = "fake-job-name"
+	experiment.StatusCheckTimeout = 10
+	var engineDetails EngineDetails
+	engineDetails.Name = "Fake Engine"
+	engineDetails.EngineNamespace = "Fake NameSpace"
+
+	tests := map[string]struct {
+		chaosengine     *litmuschaosv1alpha1.ChaosEngine
+		chaosexperiment *litmuschaosv1alpha1.ChaosExperiment
+		configmap       v1.ConfigMap
+		isErr           bool
+	}{
+		"Test Positive-1": {
+			chaosengine: &litmuschaosv1alpha1.ChaosEngine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      engineDetails.Name,
+					Namespace: engineDetails.EngineNamespace,
+				},
+				Spec: litmuschaosv1alpha1.ChaosEngineSpec{
+					Experiments: []litmuschaosv1alpha1.ExperimentList{
+						{
+							Name: experiment.Name,
+							Spec: v1alpha1.ExperimentAttributes{
+								Components: v1alpha1.ExperimentComponents{
+									ConfigMaps: []v1alpha1.ConfigMap{
+										{
+											Name:      fakeConfigMap,
+											MountPath: "fake mountpath",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			chaosexperiment: &litmuschaosv1alpha1.ChaosExperiment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      experiment.Name,
+					Namespace: experiment.Namespace,
+				},
+				Spec: litmuschaosv1alpha1.ChaosExperimentSpec{
+					Definition: litmuschaosv1alpha1.ExperimentDef{
+						Image: fakeExperimentImage,
+					},
+				},
+			},
+			configmap: v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fakeConfigMap,
+					Namespace: experiment.Namespace,
+				},
+				Data: map[string]string{
+					"my-fake-key": "myfake-val",
+				}},
+			isErr: false,
+		},
+		"Test Negative-1": {
+			chaosengine: &litmuschaosv1alpha1.ChaosEngine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      engineDetails.Name,
+					Namespace: engineDetails.EngineNamespace,
+				},
+				Spec: litmuschaosv1alpha1.ChaosEngineSpec{
+					Experiments: []litmuschaosv1alpha1.ExperimentList{
+						{
+							Name: experiment.Name,
+							Spec: v1alpha1.ExperimentAttributes{
+								Components: v1alpha1.ExperimentComponents{
+									ConfigMaps: []v1alpha1.ConfigMap{
+										{
+											Name:      fakeConfigMap,
+											MountPath: "fake mountpath",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			 chaosexperiment: &litmuschaosv1alpha1.ChaosExperiment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      experiment.Name,
+					Namespace: experiment.Namespace,
+				},
+				Spec: litmuschaosv1alpha1.ChaosExperimentSpec{
+					Definition: litmuschaosv1alpha1.ExperimentDef{
+						Image: fakeExperimentImage,
+					},
+				},
+			},
+			isErr: true,
+		},
+	}
+
+	for name, mock := range tests {
+		t.Run(name, func(t *testing.T) {
+			client := CreateFakeClient(t)
+
+			_, err := client.KubeClient.CoreV1().ConfigMaps(experiment.Namespace).Create(&mock.configmap)
+			if err != nil {
+				t.Fatalf("configmap not created, err: %v", err)
```

Comment:
would you like to add more details inside failed msg? so that from the logs we can find the failed experiment details directly

check other places as well